satpy/readers/seviri_l1b_native.py

```diff
@@ -480,6 +521,85 @@ def get_dataset(self, dataset_id, dataset_info):
 
         return dataset
 
+    def is_roi(self):
+        """Check if data covers a selected region of interest (ROI), rather than the default FES or RSS regions."""
+        is_rapid_scan = self.trailer['15TRAILER']['ImageProductionStats']['ActualScanningSummary']['ReducedScan']
+
+        # Standard RSS data is assumed to cover the three northmost segements, thus consisting of all 3712 columns and
+        # the 1392 northmost lines
+        sec15hd = self.header['15_SECONDARY_PRODUCT_HEADER']
+        north = int(sec15hd['NorthLineSelectedRectangle']['Value'])
+        east = int(sec15hd['EastColumnSelectedRectangle']['Value'])
+        south = int(sec15hd['SouthLineSelectedRectangle']['Value'])
+        west = int(sec15hd['WestColumnSelectedRectangle']['Value'])
+        ncolumns = west - east + 1
+
+        is_top3segments = (ncolumns == VISIR_NUM_COLUMNS) and \
+                          (north == VISIR_NUM_LINES) and \
+                          (south == 5 / 8 * VISIR_NUM_LINES + 1)
+
+        return not self.mda['is_full_disk'] and not (is_rapid_scan and is_top3segments)
+
+    def pad_hrv_data(self, dataset):
+        """Pad HRV data with empty pixels."""
+        logger.debug('Padding HRV data to full disk')
+
+        nlines = int(self.mda['hrv_number_of_lines'])
+        ncols = int(self.mda['hrv_number_of_columns'])
+
+        if not self.is_roi():
```

Comment:
This could also be subdivided further to two method, one for ROI and one for the other cases.

Reply:
Yes, good point! I'll split this in a similar manner as discussed above :)